tests/providers/integration_tests.py

```diff
@@ -233,12 +234,14 @@ def test_Provider_when_calling_update_record_with_fqdn_name_should_modify_record
             provider.authenticate()
             assert provider.create_record('TXT','orig.testfqdn.{0}.'.format(self.domain),'challengetoken')
             records = provider.list_records('TXT','orig.testfqdn.{0}.'.format(self.domain))
-            assert provider.update_record(records[0]['id'],'TXT','updated.testfqdn.{0}.'.format(self.domain),'challengetoken')
+            assert provider.update_record(records[0].get('id', None),'TXT','updated.testfqdn.{0}.'.format(self.domain),'challengetoken')
 
-    ###########################################################################
-    # Provider.delete_record()
-    ###########################################################################
+    # ###########################################################################
+    # # Provider.delete_record()
+    # ###########################################################################
     def test_Provider_when_calling_delete_record_by_identifier_should_remove_record(self):
+        if self.provider_name in ['route53']:
```

Comment:
don't skip tests for a specific provider in the test. That can be done in the provider test wrapper. See PointHQ: https://github.com/AnalogJ/lexicon/blob/master/tests/providers/test_pointhq.py#L18